### PR TITLE
[REG-1099] GitHub Action Support / Example

### PR DESCRIPTION
**Pull request by vontell**
This PR serves as an example for CI/CD support for Regression Games in GitHub Actions.

Because of the way this pipeline was setup, some of the code is actually already in main.

See also (https://github.com/Regression-Games/RGUnityBots/pull/31)

### CODE REVIEW

File: Assets/Tests/RGBotTests.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+{
+    [UnityTest]
+    public IEnumerator RunBotTest()
+    {
+
+        // Override this to change how long a test will wait for bots to join before failing
+        const int TIMEOUT_IN_SECONDS = 300;
+
+        // For in-editor purposes, feel free to define a default bot to use!
+        int defaultBotId = 109;
```
Comment (RG-nAmKcAz) [inline Assets/Tests/RGBotTests.cs:21]: this value seems arbitrary or specific to 1 developer's specific environment

**Review by RG-nAmKcAz — CHANGES_REQUESTED** [review]
What are the new `InitTestScene#####` files for ?

Are these intentional, or something that should be in .gitignore ?

**Review by RG-nAmKcAz — APPROVED** [review]

**Review by RG-nAmKcAz — CHANGES_REQUESTED** [review]
Marking request changes so that we don't accidentally merge without updating the branch after merging PR (https://github.com/Regression-Games/RGUnityBots/pull/31)

**Comment by RG-nAmKcAz** [comment]
@abeizer you can mark this approved, Have confirmed this is now working 

(Test Result: https://github.com/Regression-Games/RGUnitySample/actions/runs/6110621216)

**Review by RG-nAmKcAz — APPROVED** [review]
@vontell Version is now updated... Approved assuming the pipeline passes 
Looks like you'll still need @abeizer to approve though

**Review by abeizer — APPROVED** [review]

**Comment by vontell** [comment]
@RG-nAmKcAz heads up I edited the gitignore and also removed some imports and hardcoded things in the test file, check my latest commit if you'd like before I merge this in!

**Comment by RG-nAmKcAz** [comment]
> @RG-nAmKcAz heads up I edited the gitignore and also removed some imports and hardcoded things in the test file, check my latest commit if you'd like before I merge this in!

You can merge it, but the pipelines fail/hang now without those settings ... oh well

**Comment by RG-nAmKcAz** [comment]
> > @RG-nAmKcAz heads up I edited the gitignore and also removed some imports and hardcoded things in the test file, check my latest commit if you'd like before I merge this in!
> 
> You can merge it, but the pipelines fail/hang now without those settings ... oh well

Nevermind.. the failure was a github infrastructure failure, not ours

**Pull request merged**